packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1064,6 +1110,158 @@ describe("BnsConnection", () => {
+        throw new Error("Expected send transaction");
+      }
+      expect(searchResultTransaction.memo).toEqual(memo);
+
+      connection.disconnect();
+    });
+
+    it("can wait for a future transaction", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+      const chainId = connection.chainId();
+
+      const { profile, mainWalletId, faucet } = await userProfileWithFaucet(chainId);
+
+      const memo = `Payment ${Math.random()}`;
+      const sendTx: SendTransaction = {
+        kind: "bcp/send",
+        creator: faucet,
+        recipient: await randomBnsAddress(),
+        memo: memo,
+        amount: defaultAmount,
+      };
+
+      const nonce = await connection.getNonce({ pubkey: faucet.pubkey });
+      const signed = await profile.signTransaction(mainWalletId, faucet, sendTx, bnsCodec, nonce);
+      const response = await connection.postTx(bnsCodec.bytesToPost(signed));
+      const transactionIdToSearch = response.transactionId;
+      const pendingResult = toListPromise(connection.liveTx({ id: transactionIdToSearch }), 1);
+
+      // no need to await block state, just wait for transaction
+      const result = (await pendingResult)[0];
+
+      if (!isConfirmedTransaction(result)) {
+        throw new Error("Expected confirmed transaction");
+      }
+      const searchResultTransaction = result.transaction;
+      expect(result.transactionId).toEqual(transactionIdToSearch);
+      if (!isSendTransaction(searchResultTransaction)) {
+        throw new Error("Expected send transaction");
+      }
+      expect(searchResultTransaction.memo).toEqual(memo);
+
+      connection.disconnect();
+    });
+
+    it("reports DeliverTx error for an existing transaction", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+      const chainId = connection.chainId();
+
+      const { profile, mainWalletId } = await userProfileWithFaucet(chainId);
+      // this will never have tokens, but can try to sign
+      const brokeIdentity = await profile.createIdentity(mainWalletId, chainId, HdPaths.simpleAddress(1234));
+
+      const sendTx: SendTransaction = {
+        kind: "bcp/send",
+        creator: brokeIdentity,
+        recipient: await randomBnsAddress(),
+        memo: "Sending from empty",
+        amount: defaultAmount,
+      };
+
+      const nonce = await connection.getNonce({ pubkey: brokeIdentity.pubkey });
+      const signed = await profile.signTransaction(mainWalletId, brokeIdentity, sendTx, bnsCodec, nonce);
+      const response = await connection.postTx(bnsCodec.bytesToPost(signed));
+      const transactionIdToSearch = response.transactionId;
+      await response.blockInfo.waitFor(info => info.state !== TransactionState.Pending);
+
+      await tendermintSearchIndexUpdated();
+
+      const result = (await toListPromise(connection.liveTx({ id: transactionIdToSearch }), 1))[0];
```

Comment:
Just another convenience function we might want:
`await firstEvent(stream)` 
to be the same as
`(await toListPromise(stream, 1)[0])`

Reply:
Not cuz it is hard, but to reduce typos, etc for clients.
No stress, I can add all helpers in wallet-demo if you don't like them in iov-core. Just proposing.